### PR TITLE
Widescreen Mode

### PR DESCRIPTION
**Pull request by nevimer**
Widescreen mode added. Some lines ported from TG. 
dreamseeker_X2Hhks4sjA

:cl:
code: Widescreen mode.
:/cl:

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -495,4 +495,5 @@ client/verb/character_setup()
 
 	to_chat(src, "You will now [(is_preference_enabled(/datum/client_preference/overhead_chat)) ? "see" : "not see"] overhead chat messages..")
 
-	feedback_add_details("admin_verb","TOHChat") */
\ No newline at end of file
+	feedback_add_details("admin_verb","TOHChat") */
+
```
Comment (Codingale) [inline code/modules/client/client_procs.dm]: Spaaaace. How does this happen to everyone lol

File: code/modules/client/widescreen.dm [hunk 2]
```diff
@@ -0,0 +1,135 @@
+/client/proc/change_view(new_size)
+	if (isnull(new_size))
+		CRASH("change_view called without argument.")
+
+	view = new_size
+	mob.reload_fullscreen()
+	if (isliving(mob))
+		var/mob/living/M = mob
+		M.update_damage_hud()
+	addtimer(CALLBACK(src,.verb/fit_viewport,10)) //Delayed to avoid wingets from Login calls.
```
Comment (Codingale) [inline code/modules/client/widescreen.dm]: Wouldn't it be easier to just call this outside of the Login proc?

I'm not sure what a winget is, as I'm not familiar with Byond internals.

File: code/modules/mob/login.dm [hunk 3]
```diff
@@ -57,7 +57,11 @@
 	var/client/my_client // Need to keep track of this ourselves, since by the time Logout() is called the client has already been nulled
 
 /mob/Login()
-
+	if(client) // Let's give us widescreen again and again.
+		if(client.view_size)
+			client.view_size.resetToDefault() // Resets the client.view in case it was changed.
+		else
+			client.change_view("21x15")
```
Comment (Codingale) [inline code/modules/mob/login.dm:64]: Is there not a toggle or other check for this by the way? It's forcefully enabled by default?

**Review by Codingale — COMMENTED** [review]
Is there not a way to toggle it for those who don't want it? Looking at it in non-widescreen monitors just moves the bars.

On top of that, if you have less than a 1440p monitor it just takes a lot of space for icons to be semi-usable. 1080p is passable; 720p is very bad.

This also breaks the size selector, and binoculars are not modified to work with this, but that's just a quick look at 5 AM after a long sleep deprived day. Will do more testing later probably.
image

File: code/modules/client/widescreen.dm (hunk 2 above)
Comment (nevimer) [reply code/modules/client/widescreen.dm]: That's a comment from TG, I am not too sure either, pretty sure it's to call the fit viewport verb once the windows loaded.

File: code/modules/mob/login.dm (hunk 3 above)
Comment (nevimer) [reply code/modules/mob/login.dm:64]: I have added preferences and toggles in my next commit!

File: code/modules/client/widescreen.dm [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+				_x = -offset
+		animate(chief, pixel_x = world.icon_size*_x, pixel_y = world.icon_size*_y, 0, FALSE, LINEAR_EASING, ANIMATION_END_NOW)
+	//Ready for this one?
+	setTo(radius)
+
+/proc/getScreenSize(widescreen)
+	return "21x15"
+
+
+// /datum/view_data/proc/resetFormat()//Cuck
```
Comment (nevimer) [inline code/modules/client/widescreen.dm]: grr tg comments

File: code/modules/client/widescreen.dm (hunk 2 above)
Comment (nevimer) [reply code/modules/client/widescreen.dm]: > Wouldn't it be easier to just call this outside of the Login proc?
> 
> I'm not sure what a winget is, as I'm not familiar with Byond internals.

So this is a good spot to call it as it reliably runs every time you connect. The other connection procs only run once, to do things like IP lookups and load slots... this one just runs every time.

**Comment by nevimer** [comment]
I've come up with something better! Closing this because of that.

**Pull request closed without merging**